cdap-api-common/src/main/java/io/cdap/cdap/api/data/schema/Schema.java

```diff
@@ -283,7 +300,29 @@ public static Schema of(Type type) {
    * @return A {@link Schema} with the given logical type.
    */
   public static Schema of(LogicalType logicalType) {
-    return new Schema(logicalType.type, logicalType, null, null, null, null, null, null, null);
+    return new Schema(logicalType.type, logicalType, null, null, null, null, null, null, null, 0, 0);
```

Comment:
this would allow somebody to create a decimal schema with 0 precision and 0 scale. To ensure that the check is always made, it's probably better to do the precision check in the constructor if the logical type is decimal, rather than having the check in the decimalOf() method.